manifests/plugin.pp

```diff
@@ -32,8 +32,22 @@
 ) {
   include ::jenkins::params
 
-  $plugin            = "${name}.hpi"
-  $plugin_parent_dir = inline_template('<%= @plugin_dir.split(\'/\')[0..-2].join(\'/\') %>')
+  $plugin     = "${name}.hpi"
+  $homedir    = $::jenkins::params::homedir
+  $jenkinsdir = $::jenkins::params::jenkinsdir
+  $jusername = $username ? {
+    ''      => $::jenkins::params::jenkins_user,
+    default => $username,
+  }
+  $jgroup = $group ? {
+    ''      => $::jenkins::params::jenkins_group,
+    default => $group,
+  }
+  $jplugin_dir = $plugin_dir ? {
+    ''      => $::jenkins::params::plugin_dir,
+    default => $plugin_dir,
+  }
```

Comment:
I had to do above dance with the parameters,
I just tried to pull in $plugin_dir = $::jenkins::params::plugin_dir
and also the others, in the parameter list of the defined type, but then all the time, the tests failed. It seemed that Puppet didn't pulled that info in at that time.

I had to set the default values for the parameters to '', because the selector didn't wanted to work when they were set to undef. 

Alternatively, I could set these variables default to undef and do check like:

if $plugin_dir {
  $jplugin_dir = $plugin_dir
} else {
  $jplugin_dir = $::jenkins::params::plugin_dir
}

if that is preferred, or let me know if there is a better way to work around that.